internal/controllers/provisioningrequest_controller.go

```diff
@@ -1189,6 +1200,52 @@ func (r *ProvisioningRequestReconciler) handleProvisioningRequestDeletion(
 	return deleteCompleted, nil
 }
 
+// handleObservabilityAddonCleanup checks for ObservabilityAddon CRs in the given namespace
+// and removes their finalizers to allow proper namespace deletion
+func (r *ProvisioningRequestReconciler) handleObservabilityAddonCleanup(ctx context.Context, namespace string) error {
+	// List ObservabilityAddon CRs in the namespace
+	observabilityAddonList := &observabilityv1beta1.ObservabilityAddonList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(namespace),
+	}
+
+	if err := r.Client.List(ctx, observabilityAddonList, listOpts...); err != nil {
+		// If the CRD doesn't exist, this is not an error - just return
+		if meta.IsNoMatchError(err) {
+			r.Logger.InfoContext(ctx, "ObservabilityAddon CRD not found, skipping cleanup",
+				slog.String("namespace", namespace))
+			return nil
+		}
+		return fmt.Errorf("failed to list ObservabilityAddon CRs in namespace %s: %w", namespace, err)
+	}
+
+	for _, addon := range observabilityAddonList.Items {
+		if len(addon.Finalizers) > 0 {
+			r.Logger.InfoContext(ctx, "Removing finalizers from ObservabilityAddon",
+				slog.String("name", addon.Name),
+				slog.String("namespace", addon.Namespace),
+				slog.Any("finalizers", addon.Finalizers))
+
+			// Remove all finalizers
+			addon.Finalizers = []string{}
```

Comment:
Shouldn't this workaround be more targeted to the problematic finalizer instead of dropping everything?